# Q&A Activity Question Week -09
    Do the following on the dataset
    1. Explain the dataset
    2. Check the summary statistics and discuss the max, min, avg, median, and percentiles.
    3. The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings.

In [9]:
import pandas as pd
import scipy.stats as stats

In [10]:
#reading data set
data = pd.read_csv('Advertising.csv')
data.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [11]:
#primary data analysis
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


## 1. Explain the dataset 

In [12]:
#imported data set
data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

#### Explanation:
   - There are 200 rows and 5 columns in the data set
   - The given dataset consists of advertisement expenses on three different medias(TV,Radio and Newspaper) and the sales.
   - Null value is not present
    

## 2. Check the summary statistics and discuss the max, min, avg, median, and percentiles.

In [13]:
data.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

#### Summary:
   - Maximum amount for advertising is spent through TV and the least is spend throug the radio.
   - Overall mean
           TV : 147.0425
           Radio : 23.2640
           Newspaper : 30.5540
   - Overall Median
           TV : 149.75
           Radio : 22.90
           Newspaper : 25.75

## 3. The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings. 

In [14]:
#ANOVA Test
f_value , p_value = stats.f_oneway(data['TV'], data['Radio'], data['Newspaper'])
print('The f value is:',f_value)
print('The p value is:',p_value)

The f value is: 358.8514595342597
The p value is: 4.552931539744962e-103


#### Summary:
- p value is < 0.05
- so, null hypothesis is rejected.
- Therefore the amount spent for three advertisements are not the same.